app/models/profile_declaration.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class ProfileDeclaration < ApplicationRecord
+  delegated_type :declarable, types: %w[EarlyCareerTeacherProfile MentorProfile]
+  belongs_to :participant_declaration
+  belongs_to :lead_provider
```

Comment:
why does this need a lead provider if participant declaration has one too?

Reply:
I had a conversation with the PM about NPQs where the association with lead provider is different and declarations can come in from multiple lead providers for the same participant. I saw this as becoming a composite primary key, but I'm probably thinking too far ahead.